### PR TITLE
[GPU][CORE] Add ov:can_be_cached Property

### LINKED ISSUE #20190: [Good First Issue][GPU]: Cannot load model when cache directory is running out of disk space
### Context

GPU plugin utilizes cache files when loading models. When the cache is full, loading another model fails with:

```bash
~/pgladkow/openvino_repro/build$ ./test_ov
terminate called after throwing an instance of 'ov::Exception'
  what():  cldnn program build failed! clCreateProgramWithBinary
Aborted 
```

This task regards solving that issue.

### What needs to be done?

The cache should be cleared when not needed. Details should be discussed with people from Contact Points section.

Steps to reproduce:

1. Get `ov.cpp` (attachment)
2. Get `CMakeLists.txt` (attachment)
3. Mount cache directory to have a limited disk space (10m)
```bash
sudo mount -t tmpfs -o size=10m tmpfs tmp_cache
```
4. Run OpenVINO app
```bash
~/pgladkow/openvino_repro/build$ cmake ../ovms
-- Configuring done
-- Generating done
-- Build files have been written to: /opt/home/k8sworker/pgladkow/openvino_repro/build
~/pgladkow/openvino_repro/build$ cmake --build .
[100%] Built target test_ov
~/pgladkow/openvino_repro/build$ ./test_ov
Hello 
```
5. Verify that cache directory is full
```bash
~/pgladkow/openvino_repro$ ls tmp_cache/
```
6. Run OV app once again - it should fail this time
```bash
~/pgladkow/openvino_repro/build$ ./test_ov
terminate called after throwing an instance of 'ov::Exception'
  what():  cldnn program build failed! clCreateProgramWithBinary
Aborted 
```

Directory structure from the original reproducer:
```bash
~/pgladkow/openvino_repro$ tree
.
├── build
│   ├── CMakeCache.txt
│   ├── CMakeFiles
│   │   ├── 3.16.3
│   │   │   ├── CMakeCCompiler.cmake
│   │   │   ├── CMakeCXXCompiler.cmake
│   │   │   ├── CMakeDetermineCompilerABI_C.bin
│   │   │   ├── CMakeDetermineCompilerABI_CXX.bin
│   │   │   ├── CMakeSystem.cmake
│   │   │   ├── CompilerIdC
│   │   │   │   ├── a.out
│   │   │   │   ├── CMakeCCompilerId.c
│   │   │   │   └── tmp
│   │   │   └── CompilerIdCXX
│   │   │       ├── a.out
│   │   │       ├── CMakeCXXCompilerId.cpp
│   │   │       └── tmp
│   │   ├── cmake.check_cache
│   │   ├── CMakeDirectoryInformation.cmake
│   │   ├── CMakeError.log
│   │   ├── CMakeOutput.log
│   │   ├── CMakeTmp
│   │   ├── Makefile2
│   │   ├── Makefile.cmake
│   │   ├── progress.marks
│   │   ├── TargetDirectories.txt
│   │   └── test_ov.dir
│   │       ├── build.make
│   │       ├── cmake_clean.cmake
│   │       ├── CXX.includecache
│   │       ├── DependInfo.cmake
│   │       ├── depend.internal
│   │       ├── depend.make
│   │       ├── flags.make
│   │       ├── link.txt
│   │       ├── progress.make
│   │       └── src
│   │           └── ov.cpp.o
│   ├── cmake_install.cmake
│   ├── Makefile
│   └── test_ov
├── ovms
│   ├── CMakeLists.txt
│   └── src
│       ├── brain-tumor-segmentation-0002-2
│       │   └── 1
│       │       ├── brain-tumor-segmentation-0002.onnx
│       │       └── description.txt
│       ├── dummy
│       │   └── 1
│       │       ├── dummy.bin
│       │       └── dummy.xml
│       ├── inception-resnet-v2-tf
│       │   └── 1
│       │       ├── inception-resnet-v2-tf.bin
│       │       ├── inception-resnet-v2-tf.mapping
│       │       └── inception-resnet-v2-tf.xml
│       ├── ov.cpp
│       └── ssdlite_mobilenet_v2_ov
│           └── 1
│               ├── ssdlite_mobilenet_v2_ov.bin
│               ├── ssdlite_mobilenet_v2_ov.mapping
│               └── ssdlite_mobilenet_v2_ov.xml
└── tmp_cache 
```
[reproducer.zip](https://github.com/openvinotoolkit/openvino/files/12782876/reproducer.zip)

### Resources

- [What is OpenVINO?](https://github.com/openvinotoolkit/openvino#what-is-openvino-toolkit)
- [Contribution guide](https://github.com/openvinotoolkit/openvino/blob/master/CONTRIBUTING.md)
- [Blog post on contributing to OpenVINO](https://github.com/openvinotoolkit/openvino/blob/master/CONTRIBUTING.md)
- [User documentation](https://docs.openvino.ai/)

### Contact points

@pgladkows 
@vladimir-paramuzov

Ticket: 104958

### PR DESCRIPTION
**Pull request by siddhant-0707**
### Details:
 - Added `ov:can_be_cached` property to `ov::CompiledModel`
 - `GPU` plugin computes available disk space and decides `ov:can_be_cached ` value

### Tickets:
 - Closes #20190

### CODE REVIEW

**Comment by siddhant-0707** [comment]
WIP, need a little help with linking the Property with `get_can_be_cached()` function!

**Comment by akuporos** [comment]
Please, add also import [here](https://github.com/openvinotoolkit/openvino/blob/master/src/bindings/python/src/openvino/runtime/properties/__init__.py#L9)

File: src/plugins/intel_gpu/src/plugin/plugin.cpp [hunk 1]
```diff
@@ -817,6 +819,33 @@ uint32_t Plugin::get_optimal_batch_size(const ov::AnyMap& options) const {
...
+        // Calculate the size of the model in bytes
+        required_disk_space = model->get_model_size(shapes);
+    } catch (...) {
+        GPU_DEBUG_INFO << "[GET_IS_CACHEABLE] Error calculating model size" << std::endl;
+        return false;
+    }
+
+    // Check the available disk space
+    struct statvfs buffer;
+    if (statvfs(cache_dir.c_str(), &buffer) != 0) {
```
Comment (yeonbok) [inline src/plugins/intel_gpu/src/plugin/plugin.cpp:839]: Isn't this OS specific?

**Comment by p-durandin** [comment]
@siddhant-0707 please resolve the conflicts

**Pull request closed without merging**